packages/test/test-end-to-end-tests/src/test/orderSequentially.spec.ts

```diff
@@ -75,19 +79,19 @@ describeCompat("Multiple DDS orderSequentially", "NoCompat", (getTestObjectProvi
 		};
 		container = await provider.makeTestContainer(configWithFeatureGates);
 		dataObject = (await container.getEntryPoint()) as ITestFluidObject;
-		sharedString = await dataObject.getSharedObject<SharedString>(stringId);
-		sharedString2 = await dataObject.getSharedObject<SharedString>(string2Id);
+		sharedString = await dataObject.getSharedObject<ISharedString>(stringId);
+		sharedString2 = await dataObject.getSharedObject<ISharedString>(string2Id);
 		sharedDir = await dataObject.getSharedObject<SharedDirectory>(dirId);
 		sharedCell = await dataObject.getSharedObject<ISharedCell>(cellId);
 		sharedMap = await dataObject.getSharedObject<ISharedMap>(mapId);
 
 		containerRuntime = dataObject.context.containerRuntime as ContainerRuntime;
 		changedEventData = [];
-		sharedString.on("sequenceDelta", (changed, _local, _target) => {
-			changedEventData.push(changed);
+		sharedString.on("sequenceDelta", (event, _target) => {
```

Comment:
This code was wrong and should not have built, but did compile since our event emitting API (on the class) is not very type safe. The interface API is better, and thus forced this to change.

Reply:
Also an example of why should always test against the "public" API when possible and why class exposure is not preferred.